scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -633,6 +624,21 @@
         (log/error e message)
         message))))
 
+(defn unassign-all
+  "Unassigns all items in the to-unassign set with the given unassigner. Must be run with the fenzo lock held."
+  [pool-name unassigner to-unassign]
+  (doseq [{:keys [task-id hostname]} to-unassign]
+    (try
+      ; Fenzo tracks which tasks are on which nodes with assignment. To, e.g., make
+      ; sure it can distribute tasks across hosts. This instructs fenzo that a task
+      ; is no longer on a given host.
+      (log/debug "In" pool-name "pool, unassigning task"
+                 task-id "from" hostname)
+      (.call unassigner task-id hostname)
+      (catch Exception e
+        (log/error e "In" pool-name "pool, failed to unassign task"
+                   task-id "from" hostname)))))
```

Comment:
Should we wrap this whole `doseq` with a timer?

Reply:
No need. No blocking happens here, there's good logging, and the resulting metric isn't interesting, as it would depend on the batch size which is a function of Cook load and time between match cycles.

Reply:
@scrosby Thoughts on this?